packages/manager/apps/telecom/src/app/telecom/telephony/line/phone/attachLine/telephony-line-phone-attach.controller.js

```diff
@@ -0,0 +1,106 @@
+import assign from 'lodash/assign';
+import get from 'lodash/get';
+
+export default class TelephonyLinePhoneOrderAttachCtrl {
+  /* @ngInject */
+  constructor(
+    $q,
+    $translate,
+    OvhApiTelephony,
+    TelephonyMediator,
+    TucIpAddress,
+    TucToast,
+    TucToastError,
+  ) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhApiTelephony = OvhApiTelephony;
+    this.TelephonyMediator = TelephonyMediator;
+    this.TucIpAddress = TucIpAddress;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+  }
+
+  $onInit() {
+    this.isLoading = true;
+    this.macAddress = null;
+    this.line = null;
+
+    this.init();
+  }
+
+  init() {
+    this.TelephonyMediator.getGroup(this.billingAccount)
+      .then((group) => {
+        this.line = group.getLine(this.serviceName);
+      })
+      .then(() =>
+        this.OvhApiTelephony.Line()
+          .v6()
+          .get({
+            billingAccount: this.line.billingAccount,
+            serviceName: this.line.serviceName,
+          })
+          .$promise.then((result) => {
+            assign(this.line, {
+              isAttachedToOtherLinesPhone: result.isAttachedToOtherLinesPhone,
+            });
+          }),
+      )
+      .then((phone) => {
+        this.phone = phone;
+      })
+      .catch((err) => new this.TucToastError(err))
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
+
+  detachPhone() {
+    this.isDetaching = true;
+    this.OvhApiTelephony.Line()
+      .v6()
+      .dissociateDevice(
+        {
+          billingAccount: this.billingAccount,
+          serviceName: this.serviceName,
+        },
+        {
+          ipAddress: this.attachedPhoneIpAddress,
+          macAddress: this.phone.macAddress,
+        },
+      )
+      .$promise.then(() => {
+        this.TucToast.success(
+          this.$translate.instant(
+            'telephony_line_phone_order_detach_device_success',
+          ),
+        );
+
+        // Cache reset
+        this.OvhApiTelephony.Line()
+          .v6()
+          .resetAllCache();
+        this.OvhApiTelephony.Line()
+          .Phone()
+          .v6()
+          .resetAllCache();
+        this.TelephonyMediator.resetAllCache();
+        this.init();
```

Comment:
No need to return the init Promise?

Reply:
I don't think so.

Reply:
@Steffy29 we recommend to return each promise, to avoid some backwards effects if some code is executed after this kind of calls. 
Plus, it indicates clearly that a Promise is made here, so the next developer is aware that some asynchronous code is executed, without having to look into the call.